server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -189,6 +191,8 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         builders.put(Names.REFRESH, new ScalingExecutorBuilder(Names.REFRESH, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5)));
         builders.put(Names.WARMER, new ScalingExecutorBuilder(Names.WARMER, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)));
         builders.put(Names.SNAPSHOT, new ScalingExecutorBuilder(Names.SNAPSHOT, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)));
+        builders.put(Names.SNAPSHOT_META, new ScalingExecutorBuilder(Names.SNAPSHOT_META, 1, 2 * allocatedProcessors,
```

Comment:
I only used 5 min here because we're using it everywhere else for now. It seems like this should really be less.
Twice the allocated processors seemed like a reasonable guess here since this is exclusively going to run on master nodes pretty much that don't have endless cores available. This might be a little low though in some cases. We could also go for an absolute value here I guess but in the end these threads will also be doing some heavy deserialisation work here and there so going way beyond the CPU count is questionable. OTOH depending on the repo implementation an upper bound may make sense too due to e.g. S3 SDK connection limits.

=> Happy to hear opinions here or discuss this :)

Reply:
I have the feeling that having a fixed default upper bound would make it easier to reason about the snapshots related thread pools. Something like `halfProcMaxAt5` maybe, so that we don't exceed the number of CPUs ad we don't either exceed by too much the default connection pool size.